### PR TITLE
[WT-449] Add a tag when an admin comps a ticket to distinguish the department the admin is from/comping the ticket

### LINKED ISSUE #449: Add a tag when an admin comps a ticket to distinguish the department the admin is from/comping the ticket on behalf of
At PPH, each department is allocated a certain number of comp tickets each season. Wondertix will need a way to track the same information. An initial suggestion is to add a tag to comped tickets that is manually entered when creating/purchasing the ticket and save this along with the ticket.

### PR DESCRIPTION
**Pull request by RuskiDre1**
### Description
UI and backend changes for complimentary ticket changes to include which department is comping a ticket. At the moment, the department options are hard-coded in a single `const` object.

image
image

### Risks
No additional risks yet

### Validation
Manual Testing

### Issue
closes #449 

### Operating System
macOS M2

### CODE REVIEW

File: client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx [hunk 1]
```diff
@@ -478,14 +492,38 @@ const AdminPurchase = () => {
...
-        >
-          <TrashCanIcon className='h-5 w-5' strokeWidth={2} />
-        </button>
+        !params.row.complimentary ? (
+          <button
+            className='bg-red-500 px-2 py-1 text-white rounded-xl hover:bg-red-600 disabled:opacity-40 m-2'
+            onClick={() => removeRow(params.row.id)}
+          >
+            Remove
+          </button>
```
Comment (henrykaus) [inline client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx:501]: It looks like you may have accidentally removed a recent change when fixing the merge conflicts. Do you think you could replace the "Remove" button with this icon button that currently exists on `main`.
```
<button
  className='p-2 rounded-lg text-zinc-500 hover:text-red-600 hover:bg-red-100
  focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500'
  onClick={() => removeRow(params.row.id)}
  aria-label='Delete ticket'
>
  <TrashCanIcon className='h-5 w-5' strokeWidth={2} />
</button>
```

File: client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx [hunk 2]
```diff
@@ -478,14 +492,38 @@ const AdminPurchase = () => {
...
+      headerName: '',
+      width: 150,
+      renderCell: (params) => (
+        params.row.complimentary ? (
+          <button
+            className='bg-red-500 px-2 py-1 text-white rounded-xl hover:bg-red-600 disabled:opacity-40 m-2'
+            onClick={() => removeRow(params.row.id)}
+          >
+            Remove
+          </button>
```
Comment (henrykaus) [inline client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx:525]: Here as well

Comment (ncallon503) [reply client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx]: Hey Henry, just changed it to the old trash icon and tested it and it works, ready to merge I think if you are ready.
Screenshot_839

**Review by henrykaus — COMMENTED** [review]
It looks like the front-end logic looks good, but there isn't anything on the backend actually storing the new field. We recently had a backend refactor, so you'll want to merge in those changes to this branch. And I imagine that the `department` column should be housed within the new `orderticketitems` table and then the logic within the `/checkout` endpoint of `server/controllers/eventController.ts` should be updated so that it actually saves the value of `department` to the db.

**Review by henrykaus — COMMENTED** [review]
One smaller issue I noticed is that when the comp checkbox is unclicked, the value in the department select input isn't reset. This may cause problems, if an admin clicks `Comp`, then selects a value, then unclicks the `comp` checkbox. In short, when comp is unselected, the department should be reset.

**Review by henrykaus — COMMENTED** [review]
@RuskiDre1 @ncallon503 Is this PR still in the works?

**Review by henrykaus — COMMENTED** [review]
@ben16williams @benfaustin95 This PR is ready for re-review

**Review by benfaustin95 — APPROVED** [review]

**Comment by benfaustin95** [comment]
Looks good!! I like the addition of the tag to the order information!

**Pull request merged**